### PR TITLE
Agg

### PR DESCRIPTION
**Pull request by Mohsendream**
Tasks :
add :
1) silver & gold packages
2)Aggregations
3)Main

### CODE REVIEW

File: src/main/scala/com/philcoutinho/gold/CommentsInTime.scala [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+import org.apache.spark.sql.{DataFrame, SparkSession}
+
+
+object CommentsInTime {
+
+  def commentsintime(CommentData : DataFrame, spark : SparkSession):DataFrame={
+
+    CommentData.createOrReplaceTempView("my_table")
+    val result = spark.sql("SELECT __typename, created_at, comment_id, owner_id, comment_username, text " +
+      " FROM my_table WHERE FROM_UNIXTIME (created_at) BETWEEN ('2020-01-01') AND ('2021-05-16')")
```
Comment (saifeddine1992) [inline src/main/scala/com/philcoutinho/gold/CommentsInTime.scala]: add the time as arguments in the function commentsintime

File: src/main/scala/com/philcoutinho/gold/CommentsInTime.scala [hunk 2]
```diff
@@ -0,0 +1,16 @@
+package com.philcoutinho.gold
+
+
+import org.apache.spark.sql.{DataFrame, SparkSession}
+
+
+object CommentsInTime {
+
+  def commentsintime(CommentData : DataFrame, spark : SparkSession):DataFrame={
```
Comment (saifeddine1992) [inline src/main/scala/com/philcoutinho/gold/CommentsInTime.scala]: commentsInTime*

File: src/main/scala/com/philcoutinho/gold/MediaCommentCount.scala [hunk 3]
```diff
@@ -0,0 +1,12 @@
+package com.philcoutinho.gold
+
+import org.apache.spark.sql.{DataFrame, SparkSession}
+
+object MediaCommentCount {
+
+  def mediacommentcount(EdgeMediaComment: DataFrame, spark: SparkSession): DataFrame = {
```
Comment (saifeddine1992) [inline src/main/scala/com/philcoutinho/gold/MediaCommentCount.scala]: mediaCommentCount

File: src/main/scala/com/philcoutinho/main.scala [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+object main {
+
+  def main(args: Array[String]): Unit = {
+    val spark = SparkSession
+      .builder()
+      .master("local[*]")
+      .appName("main")
+      .getOrCreate()
+    val path = "C:\\Users\\user\\IdeaProjects\\PhilCoutinho_2\\target\\sample.json"
+    val outpath = "C:\\Users\\user\\IdeaProjects\\PhilCoutinho_2\\target\\"
```
Comment (saifeddine1992) [inline src/main/scala/com/philcoutinho/main.scala]: if you put parquet tables under the target directory, your tables will be pushed if u push the target directory: which shouldn't be the case

File: src/main/scala/com/philcoutinho/gold/CommentsInTime.scala [hunk 5]
```diff
@@ -0,0 +1,16 @@
+package com.philcoutinho.gold
+
+
```
Comment (saifeddine1992) [inline src/main/scala/com/philcoutinho/gold/CommentsInTime.scala]: spaces

File: src/main/scala/com/philcoutinho/gold/CommentsInTime.scala [hunk 6]
```diff
@@ -0,0 +1,16 @@
...
+import org.apache.spark.sql.{DataFrame, SparkSession}
+
+
```
Comment (saifeddine1992) [inline src/main/scala/com/philcoutinho/gold/CommentsInTime.scala]: space

File: src/main/scala/com/philcoutinho/gold/MediaCommentCount.scala [hunk 7]
```diff
@@ -0,0 +1,12 @@
+package com.philcoutinho.gold
+
```
Comment (saifeddine1992) [inline src/main/scala/com/philcoutinho/gold/MediaCommentCount.scala:2]: remove redundant spaces... apply to entire project

File: src/main/scala/com/philcoutinho/silver/Comments.scala [hunk 8]
```diff
@@ -0,0 +1,22 @@
...
+import org.apache.spark.sql.DataFrame
+import org.apache.spark.sql.functions.{col, explode}
+
+object Comments {
+
+  def comments(InputData: DataFrame): DataFrame = {
+    val explodedGraphImage = InputData.select(explode(col("GraphImages")).as("GraphImage"))
+    val explodedDataArray = explodedGraphImage.select(explode(col("GraphImage.comments.data")).as("Data"),
+      col("GraphImage.username").as("username"),
+      col("GraphImage.__typename").as("__typename"))
```
Comment (saifeddine1992) [inline src/main/scala/com/philcoutinho/silver/Comments.scala:13]: change "__typename" to "typename"

**Review by saifeddine1992 — CHANGES_REQUESTED** [review]
add aggregations : 
- likes over time ( a dataframe that contains three columns, one for likes, the second for dates and the third for post_owner(cristizno, messi ...etc))
- most active influencers ( the top 10 most active influencers on instagram, which is the number of posts per week)

File: src/main/scala/com/philcoutinho/gold/CommentsInTime.scala (hunk 2 above)
Comment (Mohsendream) [reply src/main/scala/com/philcoutinho/gold/CommentsInTime.scala]: done

File: src/main/scala/com/philcoutinho/gold/MediaCommentCount.scala (hunk 3 above)
Comment (Mohsendream) [reply src/main/scala/com/philcoutinho/gold/MediaCommentCount.scala]: done

File: src/main/scala/com/philcoutinho/main.scala (hunk 4 above)
Comment (Mohsendream) [reply src/main/scala/com/philcoutinho/main.scala]: done

File: src/main/scala/com/philcoutinho/gold/CommentsInTime.scala (hunk 5 above)
Comment (Mohsendream) [reply src/main/scala/com/philcoutinho/gold/CommentsInTime.scala]: done

File: src/main/scala/com/philcoutinho/gold/CommentsInTime.scala (hunk 6 above)
Comment (Mohsendream) [reply src/main/scala/com/philcoutinho/gold/CommentsInTime.scala]: done

File: src/main/scala/com/philcoutinho/gold/MediaCommentCount.scala (hunk 7 above)
Comment (Mohsendream) [reply src/main/scala/com/philcoutinho/gold/MediaCommentCount.scala:2]: done

File: src/main/scala/com/philcoutinho/silver/Comments.scala (hunk 8 above)
Comment (Mohsendream) [reply src/main/scala/com/philcoutinho/silver/Comments.scala:13]: done

File: src/main/scala/com/philcoutinho/gold/CommentsInTime.scala (hunk 1 above)
Comment (Mohsendream) [reply src/main/scala/com/philcoutinho/gold/CommentsInTime.scala]: Done

**Review by saifeddine1992 — APPROVED** [review]

**Pull request merged**